extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/DimmerThingHandler.java

```diff
@@ -0,0 +1,294 @@
+import org.eclipse.smarthome.binding.dmx.internal.DmxThingHandler;
+import org.eclipse.smarthome.binding.dmx.internal.Util;
+import org.eclipse.smarthome.binding.dmx.internal.ValueSet;
+import org.eclipse.smarthome.binding.dmx.internal.action.FadeAction;
+import org.eclipse.smarthome.binding.dmx.internal.multiverse.BaseChannel;
+import org.eclipse.smarthome.binding.dmx.internal.multiverse.Channel;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DimmerThingHandler} is responsible for handling commands, which are
+ * sent to the dimmer.
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+
+public class DimmerThingHandler extends DmxThingHandler {
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_DIMMER);
+
+    private final Logger logger = LoggerFactory.getLogger(DimmerThingHandler.class);
+
+    private List<Channel> channels = new ArrayList<Channel>();
+    private ValueSet turnOnValue = new ValueSet(0, -1, Channel.MAX_VALUE);
+    private ValueSet turnOffValue = new ValueSet(0, -1, Channel.MIN_VALUE);
+
+    private int fadeTime = 0, dimTime = 0;
+
+    private boolean isDimming = false;
+
+    public DimmerThingHandler(Thing dimmerThing) {
+        super(dimmerThing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Integer channelCounter = 0;
+        switch (channelUID.getId()) {
+            case CHANNEL_BRIGHTNESS:
+                if (command instanceof PercentType) {
+                    logger.trace("adding fade to channels in thing {}", this.thing.getUID());
+                    int targetValue = Util.toDmxValue((PercentType) command);
+                    channels.forEach(channel -> channel
+                            .setChannelAction(new FadeAction(fadeTime, channel.getValue(), targetValue, -1)));
+                } else if (command instanceof DecimalType) {
+                    logger.trace("adding fade to channels in thing {}", this.thing.getUID());
+                    int targetValue = ((BigDecimal) command).intValue();
```

Comment:
Eclipse complains that Command cannot be casted to Integer. Therefore I used BigDecimal, which is available.